src/main/scala/org/apache/spark/sql/delta/storage/BaseExternalLogStore.scala

```diff
@@ -90,21 +90,53 @@ abstract class BaseExternalLogStore(sparkConf: SparkConf, hadoopConf: Configurat
    * Method try to rewrite TransactionLog from temporary path if it not exists.
    * Method returns completed [[LogEntryMetadata]]
    */
-
-   private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
+  private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
     logDebug(s"Try to fix: ${entry.path}")
-    val tempPath = entry.tempPath.get
     try {
-      writeActions(fs, entry.path, read(tempPath).iterator)
-      val completedEntry = entry.complete()
-      writeCache(fs, completedEntry, overwrite = true)
-      deleteFile(fs, tempPath)
-      completedEntry
+      writeLogTransaction(fs, entry)
     } catch {
-      case e: FileNotFoundException =>
-        logWarning(s"ignoring $e while fixing (already done?)")
-        entry
+      case _: FileSystemException => print("Failed to fix transaction, ignore...")
     }
+    val completedEntry = entry.complete()
+    writeCache(fs, completedEntry, overwrite = true)
+    completedEntry
+  }
+
+  private def writeLogTransaction(fs: FileSystem, entryMetadata: LogEntryMetadata) {
+    try {
+      FileUtil.copy(
+        entryMetadata.tempPath.get.getFileSystem(getHadoopConfiguration),
+        entryMetadata.tempPath.get,
+        entryMetadata.path.getFileSystem(getHadoopConfiguration),
+        entryMetadata.path,
+        true,
+        true,
+        getHadoopConfiguration
+      )
+    } catch {
+      case e: Exception => throw new FileSystemException(
+        s"Cannot rename file from ${entryMetadata.tempPath.get} to ${entryMetadata.path}",
+        "",
+        e.getMessage
+      )
+    }
+  }
+
+
+  /**
+   * Check path exists on filesystem or in cache
+   * @param fs reference to [[FileSystem]]
+   * @param resolvedPath path to check
+   * @return Boolean true if file exists else false
+   */
+  protected def exists(
```

Comment:
it seems it is used only by test subclass now, so probably it should be moved there